.github/workflows/workspace-images.yml

```
@@ -0,0 +1,50 @@
+        if: ${{ env.CAN_PUSH_IMAGE == 'true' }}
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Check whether workspaces/${{ matrix.workspace }} was modified
+        run: |
+          tools/check_path_modified.sh workspaces/${{ matrix.workspace }} workspaces_${{ matrix.workspace }}_modified
+
+      - name: Build and push prairielearn/workspace-${{ matrix.workspace }}
+        if: env[format('workspaces_{0}_modified', matrix.grader)] == 'true'
```

actionlint (1.7.4)

property "grader" is not defined in object type {workspace: string}

(expression)